src/hooks/useChainIdBackground.ts

```diff
@@ -0,0 +1,23 @@
+import { SupportedChainId } from 'constants/chains'
+import React from 'react'
+import { useActiveWeb3React } from './web3'
+
+export const useChainIdBackground = () => {
```

Comment:
can you just make this a component? this logic does not need to be shared or ever called more than once. like call it BackgroundColorNetworkUpdater

Reply:
Yeah, I can see that and I'm fine w/ it in this case since all the other updaters use that API. I just prefer not using components that never return dom since the api surface when used tends to imply document structure; a null returning component never has that. 🤷 Hooks communicate the side effect more clearly, I think.

Reply:
hooks do not implicitly have side effects as react components do not implicitly return dom, but making it a hook implies it is to be shared

Reply:
What would you think of adding this to `src/state/application/updater.ts`?

Reply:
Seems like the existing updaters are all state-focused, and this one's kind of an oddball. Not sure where it belongs.

Reply:
don't feel strongly about where it goes